blogs/ecs-ec2-crossvpc-with-tls/appmesh-baseline.yml

```diff
@@ -0,0 +1,937 @@
+                response = ec2.create_key_pair(KeyName=keyname)
+                ssm.put_parameter(
+                  Name = paramName,
+                  Value = response['KeyMaterial'],
+                  Type = 'SecureString',
+                  KeyId = keyid,
+                  Overwrite = True
+                )
+                data['Name'] = keyname
+                cfnresponse.send(event, context, cfnresponse.SUCCESS, data)
+            except Exception as e:
+              errorMessage = e.response['Error']['Message']
+              data['Message'] = errorMessage
+              cfnresponse.send(event, context, cfnresponse.FAILED, data)
+  KeyPairHelperExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: "/"
+      Policies:
+        - PolicyName: KeyPairHelperExecutionPolicy
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow 
+                Action:
+                  - 'logs:CreateLogGroup'
+                  - 'logs:CreateLogStream'
+                  - 'logs:PutLogEvents'
+                Resource: arn:aws:logs:*:*:*
+              - Effect: Allow
+                Action:
+                  - 'ec2:CreateKeyPair'
+                  - 'ec2:DeleteKeyPair'
+                  - 'ssm:PutParameter'
+                  - 'ssm:DeleteParameter'
+                  - 'kms:Encrypt'
+                Resource: "*"
+
+  KMSKey:
+    Type: AWS::KMS::Key
+    Properties:
+      KeyPolicy:
+        Version: "2012-10-17"
+        Id: master-key
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action:
+              - 'kms:Create*'
+              - 'kms:Describe*'
+              - 'kms:Enable*'
+              - 'kms:List*'
+              - 'kms:Put*'
+              - 'kms:Update*'
+              - 'kms:Revoke*'
+              - 'kms:Disable*'
+              - 'kms:Get*'
+              - 'kms:Delete*'
+              - 'kms:TagResource'
+              - 'kms:UntagResource'
+              - 'kms:ScheduleKeyDeletion'
+              - 'kms:CancelKeyDeletion'
+            Resource: "*"
+          - Sid: Allow Use of the Key
+            Effect: Allow
+            Principal:          
+              AWS: 
+                - !GetAtt KeyPairHelperExecutionRole.Arn
+                - !Sub arn:aws:iam::${AWS::AccountId}:role/${Cloud9IAMRole}
+            Action:
+              - 'kms:Encrypt'
+              - 'kms:Decrypt'
+              - 'kms:ReEncrypt*'
+              - 'kms:GenerateDataKey*'
+              - 'kms:DescribeKey'
+            Resource: "*"
```

Comment:
Have we verified that these are indeed the least-privileges actions required for this role? At a glance it looks very broad.